New layer for all-the-icons

Stebalien opened the pull request:
This change introduces a new "all-the-icons" layer to configure all-the-icons and related packages.

This layer:

* Configures all the icons for dired, ivy, company-box, and ibuffer
* Defines icon transformers for ivy-rich but doesn't use them by default.
* Defines additional icons for some common spacemacs modes.

Stebalien commented on layers/+completion/auto-completion/packages.el:
```diff
@@ -150,59 +149,10 @@
        (unless (eq auto-completion-enable-help-tooltip 'manual)
          (company-quickhelp-mode))))))
 
-(defun auto-completion/init-all-the-icons ()
```
*Should* this be moved? I can also just change this to a pre-init hook and move ownership.

smile13241324 replied:
Hmm we need to make sure that company box continues to work for users not using all-the-icons everywhere. 

Not sure if this package works at all at the moment but nevertheless the necessary config for it must be applied even when all-the-icons-layer is not loaded.

Lets wait with this point until we have a confirmation that company-box works at all if it does not we can just keep it as it is.

Stebalien replied:
Company box should still work, this is just the part that configures company-box to use all-the-icons.

smile13241324 replied:
This should be fine then

Stebalien commented on layers/+misc/all-the-icons/README.org:
```diff
@@ -0,0 +1,73 @@
+your ivy-rich results as follows.
+
+#+BEGIN_SRC elisp
+  (setq ivy-rich-display-transformers-list
```
I'd like to configure this by default, but there isn't a clean way to edit this list without messing with the user's customizations, as far as I know.

smile13241324 replied:
I would agree, changing this in an automatic way like walking the tree and filling in functions which have not been set could have very suprising effects for many users. Lets just keep it easy we can later on optimise this point in separate PRs.

Stebalien commented on layers/+misc/all-the-icons/packages.el:
```diff
@@ -0,0 +1,135 @@
+    ;; EXWM
+    (add-to-list 'all-the-icons-mode-icon-alist
+                 '(exwm-mode all-the-icons-octicon "browser" :face all-the-icons-dsilver))
+    ;; Notmuch
```
Should these go in the notmuch layer?

smile13241324 replied:
Hmmm, I would say no everything all-the-icons specific should be centralised here. However I would be interested how your layer works if for example notmuch is not installed. It should definitely test for the installation status of these packages before trying to modify its config.

Stebalien replied:
It shouldn't be a problem if the mode isn't installed. The list contains a tons of non-default modes out of the box anyways.

Stebalien commented on layers/+misc/all-the-icons/packages.el:
```diff
@@ -0,0 +1,135 @@
+  (with-eval-after-load 'company-box
+    ;; Icons selected by liguangsheng
+    ;; https://github.com/liguangsheng/emacsd/blob/master/lisp/init-completion.el
+      (defun spacemacs/company-box-icon (family icon &rest args)
```
Renamed from `my-company-box-icon`.

smile13241324 replied:
Should be ok, company box is not working for me either with or without your layer, I have asked the others if it is working for either of them before we merge.

smile13241324 replied:
Needs testing.

Stebalien commented on layers/+misc/all-the-icons/packages.el:
```diff
@@ -0,0 +1,135 @@
+            (all-the-icons-icon-for-mode 'fundamental-mode)
+          icon)))))
+
+(defun all-the-icons/post-init-company-box ()
```
Reviewer: Unfortunately, I can't seem to get company-box working (icons aside). I don't think I broke anything here, but please double check.

smile13241324 replied:
Same here, lets see what the others say if it is not working on their side either we merge your pr and fix company box in another one.

smile13241324 replied:
I have got confirmation from @duianto that company box is working on his side. But when this is just for adding all the icon fonts it should be ok to keep it here.

I will try to test this before the merge.

nixmaniack commented:
Should this layer be auto-added if you select the mode line theme as `all-the-icons`?

Stebalien commented:
> Should this layer be auto-added if you select the mode line theme as all-the-icons?

Probably not. This configures all the icons all over the place.

smile13241324 commented on layers/+misc/all-the-icons/packages.el:
```diff
@@ -0,0 +1,135 @@
+              (TypeParameter . ,(spacemacs/company-box-icon 'faicon "cog"))
+              (Template . ,(spacemacs/company-box-icon 'octicon "file-code"))))))
+
+;;; packages.el ends here
```
No package ends here declaration in a layer file please

smile13241324 commented on layers/+misc/all-the-icons/packages.el:
```diff
@@ -0,0 +1,135 @@
+;;
+;;; License: GPLv3
+
+;;; Commentary:
```
No commentary and code block in layer files please.

smile13241324 commented on layers/+misc/all-the-icons/packages.el:
```diff
@@ -0,0 +1,135 @@
+;;; packages.el --- all-the-icons layer packages file for Spacemacs.
+;;
+;; Copyright (c) 2012-2018 Sylvain Benner & Contributors
```
This should be ;; Copyright (c) 2012-2020 Sylvain Benner & Contributors

smile13241324 commented on layers/+misc/all-the-icons/README.org:
```diff
@@ -0,0 +1,73 @@
+To use this configuration layer, add it to your =~/.spacemacs=. You will need to
+add =all-the-icons= to the existing =dotspacemacs-configuration-layers= list in
+this file.
+
```
Please provide a comment that users should install all-the-icons-font. I think there is a function available for this in emacs.

Stebalien commented:
I believe I have addressed your review. Note: I've amended my commit, is that the correct way to contribute to spacemacs? Or would you prefer additional commits and a squash at the end?

smile13241324 commented:
Amending your commits is the correct way, see CONTRIBUTING.org for details.

On Sun, 15 Mar 2020, 02:01 Steven Allen, <notifications@github.com> wrote:

> I believe I have addressed your review. Note: I've amended my commit, is
> that the correct way to contribute to spacemacs? Or would you prefer
> additional commits and a squash at the end?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/syl20bnr/spacemacs/pull/13337#issuecomment-599153906>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEKCTAGFUBTTHVWM767NKADRHQSH3ANCNFSM4K7KRWDA>
> .
>

smile13241324 commented:
Do you plan to continue working on this PR @Stebalien? If not please drop me a line and I will try to take over.

Stebalien commented:
Ah, no, sorry. And thanks for the ping.

I've switched to a custom config (wanted something a bit more minimal at this point).
